rest/rest-sqs-testing-amazon-java-sdk/src/test/scala/org/elasticmq/rest/sqs/AmazonCliTestSuite.scala

```diff
@@ -509,6 +522,21 @@ class AmazonCliTestSuite
 
       result.parseJson.convertTo[GetQueueAttributesResponse].Attributes.get("VisibilityTimeout") shouldBe Some("99")
     }
+
+    test(s"should get source queues of a dlq with ${version.name}") {
+      // given
+      val dlq = createQueue("testDlq")
+      val redrivePolicy = RedrivePolicy("testDlq", awsRegion, awsAccountId, 1).toJson.toString().replace("\"", "\\\\\\\"") // annoyingly, really does need this many escapes
+      val main = createQueue("main", Some(s"{\\\"RedrivePolicy\\\":\\\"$redrivePolicy\\\"}"))
```

Comment:
Unfortunately I can't get this to work with your suggestion, due to a few quirks with the CLI and the dead letter JSON. Error on my machine with your exact code if you're curious:
```
Error parsing parameter '--attributes': Invalid JSON: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
JSON received: {RedrivePolicy:{"deadLetterTargetArn":"arn:aws:sqs:elasticmq:123456789012:testDlq","maxReceiveCount":1}}
``` 
AFAICT this is due to the following facts:
- The shell removes unescaped quotes for some reason (so we need to pass a literal `\"RedrivePolicy\"`)
- The dead letters endpoint doesn't actually want a deep JSON object, they want a map of String -> String, and then they parse the RedrivePolicy as JSON afterwards. So due to the shell removing one set, the redrive policy JSON actually has to be double-escaped. Add in the escapes for the Scala strings and it means it must be written as triple-escaped. I reformatted `"\\\\\\\""` as `"""\\\""""` in the latest change but I'm not sure that is much better lol
- I can't even improve the bottom line full JSON by using """ because Scala interprets backslash in `s"""` blocks as escape, instead of literally (I assume so they can be used to escape `$`)

If any of this behavior is different for you or you're able to find a cleaner way to go about it then feel free to let me know or make the change, but this is the only way I was able to get it to work. 🤷